File: How_To_Program_Exercises/Visual_C#_How_To_Program2012_Fifth/Chapter5/IterationFactorials.cs
/************************************************
 *	Program that reads a nonnegative integer	*  
 * 	and computes and prints its factorial.		*
 ************************************************/
 
 using System;
 
 class IterationFactorials
{
	public static void Main (string[] args)
	{
		int inputNumber;
		int n;		
		int factorial = 1;
		
		Console.Write("\nEnter a positive integer: ");
		n = inputNumber = Convert.ToInt32(Console.ReadLine());
		
		while ( n > 0)
		{
			factorial = factorial * n;
			n = n - 1;
		}
		
		Console.WriteLine("{0}! is {1}", inputNumber, factorial);
	}
}

File: How_To_Program_Exercises/C++How_To_Program_8/Chapter4/IterationFactorials.cpp
/************************************************
 *	Program that reads a nonnegative integer	*  
 * 	and computes and prints its factorial.		*
*************************************************/
#include <iostream>

int main ()
{
	int inputNumber, n;
	int factorial = 1;
	std::cout << "\nEnter a positive integer: ";
	std::cin >> inputNumber;
	
	n = inputNumber;
	
	while ( n > 0)
	{
		factorial = factorial * n;
		n = n - 1;
	}
	
	std::cout << inputNumber << "! is " << factorial << std::endl;

	return 0;
}